src/Microsoft.TestPlatform.Common/Interfaces/Engine/ClientProtocol/IProxyDiscoveryManager.cs

```diff
@@ -28,6 +28,12 @@ public interface IProxyDiscoveryManager
     /// </summary>
     void Abort();
 
+    /// <summary>
+    /// Aborts discovery operation with EventHandler.
+    /// </summary>
+    /// <param name="eventHandler">EventHandler for handling discovery events from Engine</param>
+    void Abort(ITestDiscoveryEventsHandler2 eventHandler);
```

Comment:
I have a suggestion but I don't know if this would work with your multi-version design. We could have an internal interface implementing the public one and we add the new methods like this one on the internal layer.

Reply:
As per our discussion we will keep it as-is for now.